src/target/riscv/riscv.c

```diff
@@ -5451,9 +5449,8 @@ static void riscv_info_init(struct target *target, struct riscv_info *r)
 	r->mem_access_methods[1] = RISCV_MEM_ACCESS_SYSBUS;
 	r->mem_access_methods[2] = RISCV_MEM_ACCESS_ABSTRACT;
 
-	r->mem_access_progbuf_warn = true;
-	r->mem_access_sysbus_warn = true;
-	r->mem_access_abstract_warn = true;
+	r->mem_access_method_num = RISCV_MEM_ACCESS_MAX_METHODS_NUM;
+	memset(r->mem_access_warn, true, RISCV_MEM_ACCESS_MAX_METHODS_NUM);
```

Comment:
Detail but important:

This assumes that `bool` is 1 byte wide which need not be true (https://stackoverflow.com/a/8014188).

For that reason, please replace this by:

```suggestion
	for (unsigned int i = 0; i < r->mem_access_method_num; i++)
		r->mem_access_warn[i] = true;
```

Generally, it'd be my preference to not use `memset` for initialization of structures (or arrays) for these reasons:

- It is error-prone w.r.t. the object size
- Static analysis tools may not properly recognize this form of initialization
- When searching for assignments to structure or array items, this is easy to miss

Reply:
Addressed

Reply:
Addressed